e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js

```diff
@@ -60,132 +60,155 @@ describe("scenarios > dashboards > filters > auto apply", () => {
+        cy.findByLabelText("Auto-apply filters").should("not.be.checked");
+      });
+      filterWidget().findByText("Gadget").should("be.visible");
+      getDashboardCard().findByText("Rows 1-4 of 53").should("be.visible");
 
-    cy.log(
-      "last applied parameter values should be used when disabling auto applying filters, even if previously there were draft parameter values",
-    );
-    filterWidget().findByText("Widget").click();
-    popover().within(() => {
-      cy.findByText("Gadget").click();
-      cy.button("Update filter").click();
-    });
-    rightSidebar().within(() => {
-      cy.findByLabelText("Auto-apply filters").click();
-      cy.wait("@updateDashboard");
-      cy.findByLabelText("Auto-apply filters").should("not.be.checked");
+      cy.log("draft parameter values should be applied manually");
+      filterWidget().findByText("Gadget").click();
+      popover().within(() => {
+        cy.findByText("Widget").click();
+        cy.button("Update filter").click();
+      });
+      getDashboardCard().findByText("Rows 1-4 of 53").should("be.visible");
+      dashboardParametersContainer().within(() => {
+        cy.button("Apply").click();
+        cy.wait("@cardQuery");
+      });
+      getDashboardCard().findByText("Rows 1-4 of 107").should("be.visible");
+      cy.get("@cardQuery.all").should("have.length", 3);
+
+      cy.log(
+        "draft parameter values should be applied when enabling auto-applying filters",
+      );
+      filterWidget().findByText("2 selections").click();
+      popover().within(() => {
+        cy.findByText("Gadget").click();
+        cy.button("Update filter").click();
+      });
+      filterWidget().findByText("Widget").should("be.visible");
+      dashboardParametersContainer().button("Apply").should("be.visible");
+      rightSidebar().within(() => {
+        cy.findByLabelText("Auto-apply filters").click();
+        cy.wait("@updateDashboard");
+        cy.findByLabelText("Auto-apply filters").should("be.checked");
+      });
+      filterWidget().findByText("Widget").should("be.visible");
+      getDashboardCard().findByText("Rows 1-4 of 54").should("be.visible");
+      cy.get("@cardQuery.all").should("have.length", 4);
+
+      cy.log(
+        "last applied parameter values should be used when disabling auto applying filters, even if previously there were draft parameter values",
+      );
+      filterWidget().findByText("Widget").click();
+      popover().within(() => {
+        cy.findByText("Gadget").click();
+        cy.button("Update filter").click();
+      });
+      rightSidebar().within(() => {
+        cy.findByLabelText("Auto-apply filters").click();
+        cy.wait("@updateDashboard");
+        cy.findByLabelText("Auto-apply filters").should("not.be.checked");
+      });
+      filterWidget().findByText("2 selections").should("be.visible");
+      cy.get("@cardQuery.all").should("have.length", 5);
+
+      // metabase#31721
```

Comment:
IMO these issue tags (i.e. `metabase#XYZ`) are important enough to deserve to be printed out in test output.

This suggestion applies to all other occurrences in this PR.

```suggestion
      cy.log("it makes no extraneous API calls (metabase#31721)");
```

Reply:
good point

Reply:
Fixed in 93f76d9f34b2fc7890526398c7175b209289be51